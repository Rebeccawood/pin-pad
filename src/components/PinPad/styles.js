import styled from "styled-components";
import config from "../../lib/config";

export const Background = styled.section`
  background: url("/assets/background.jpg");
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: ${config.borderRadius.desktop};
  border: ${config.border.green};
  box-shadow: ${config.palette.grey} 3rem -3rem 3rem -1.8rem;
  height: 70rem;
  text-align: center;
  width: 80rem;

  @media (max-width: ${config.resolutions.mobileL}) {
    border-radius: ${config.borderRadius.mobile};
    height: auto;
    margin-bottom: 2rem;
    padding: 1.5rem;
    width: 100%;
  }
`;
