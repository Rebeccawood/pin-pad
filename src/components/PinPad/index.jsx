import React, { useState } from "react";
import { Background } from "./styles.js";
import Display from "../Display";
import ButtonPanel from "../ButtonPanel";

const PinPad = () => {
  const [displayNumber, setDisplayNumber] = useState([]);

  return (
    <Background>
      <Display
        setDisplayNumber={setDisplayNumber}
        displayNumber={displayNumber}
      />
      <ButtonPanel
        setDisplayNumber={setDisplayNumber}
        displayNumber={displayNumber}
      />
    </Background>
  );
};

export default PinPad;
