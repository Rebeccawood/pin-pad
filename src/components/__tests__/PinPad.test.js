import React from "react";
import PinPad from "../PinPad/index.jsx";
import "@testing-library/jest-dom/extend-expect";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("PinPad Component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<PinPad />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
