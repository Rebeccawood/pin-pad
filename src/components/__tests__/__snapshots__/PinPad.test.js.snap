// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinPad Component matches the snapshot 1`] = `
<section
  className="sc-AxjAm dmFqzf"
>
  <section
    className="sc-AxirZ glVpXs"
  >
    <section
      className="sc-AxiKw cCChKf"
    >
      <p>
        
      </p>
    </section>
  </section>
  <section
    className="sc-AxhUy inSJRW"
  >
    <section
      className="sc-AxgMl iIjfol"
    >
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        0
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        1
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        2
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        3
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        4
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        5
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        6
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        7
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        8
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        9
      </div>
      <div
        className="sc-AxheI jXaxWO"
        onClick={[Function]}
      >
        X
      </div>
    </section>
  </section>
</section>
`;
